java/demo/pom.xml

```diff
@@ -76,13 +82,64 @@
     <!-- for hot reload of the web application-->
     <outputDirectory>${project.build.directory}/${project.build.finalName}/WEB-INF/classes</outputDirectory>
     <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <version>3.1.0</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>java</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <mainClass>com.google.template.soy.SoyParseInfoGenerator</mainClass>
+                    <arguments>
+                      <argument>--generateInvocationBuilders</argument>
+                      <argument>--javaPackage</argument>
+                      <argument>com.google.phonenumbers.demo.template</argument>
+                      <argument>--javaClassNameSource</argument>
+                      <argument>filename</argument>
+                      <argument>--srcs</argument>
+                      <argument>${soy.root}/input_form.soy,${soy.root}/result.soy,${soy.root}/result_error.soy,${soy.root}/result_file.soy</argument>
+                      <argument>--outputDirectory</argument>
+                      <argument>src/main/java/com/google/phonenumbers/demo/template/</argument>
+                    </arguments>
+        </configuration>
+      </plugin>
       <plugin>
         <groupId>com.google.cloud.tools</groupId>
         <artifactId>appengine-maven-plugin</artifactId>
-        <version>1.3.1</version>
+        <version>2.4.4</version>
         <configuration>
         </configuration>
       </plugin>
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <source>8</source>
+          <target>8</target>
+        </configuration>
+        <groupId>org.apache.maven.plugins</groupId>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-javadoc-plugin</artifactId>
+        <version>2.7</version>
+        <executions>
+          <execution>
+            <id>attach-javadocs</id>
+            <goals>
+              <goal>jar</goal>
+            </goals>
+            <configuration>
+              <failOnError>false</failOnError> <!-- SoyParseInfoGenerator.jar does not seem to create valid javadoc which requires failOnError false to compile -->
```

Comment:
I'm sorry to nag, this is the last one :) Let's rather just exclude the package with generated templates: https://maven.apache.org/plugins/maven-javadoc-plugin/examples/exclude-package-names.html